### PR TITLE
add Singulars division

### PR DESCRIPTION
**Pull request by hannes14**

### CODE REVIEW

File: src/ideal/ideal.jl [hunk 1]
```diff
@@ -646,6 +646,20 @@ function reduce(a::T, b::T) where T <: SPolyUnion
    return R(ptr)
 end
 
+@doc Markdown.doc"""
+    division(I::sideal{S}, G::sideal{S}) where S <: SPolyUnion
+
+Computes a division with remainder.
+returns a list T,Rest,U where matrix(I)*matrix(U)=matrix(G)*matrix(T)+matrix(Rest)
```
Comment (fingolfin) [inline src/ideal/ideal.jl:653]:
```suggestion
Returns a tuple `T,Rem,U` where `matrix(I)*matrix(U)=matrix(G)*matrix(T)+matrix(Rem)`.
```

File: src/module/module.jl [hunk 2]
```diff
@@ -177,6 +177,20 @@ function reduce(M::smodule, G::smodule)
    return Module(R, ptr)
 end
 
+@doc Markdown.doc"""
+    division(I::smodule{S}, G::smodule{S}) where S
+
+Computes a division with remainder.
+returns a list T,Rest,U where matrix(I)*matrix(U)=matrix(G)*matrix(T)+matrix(Rest)
```
Comment (fingolfin) [inline src/module/module.jl:184]:
```suggestion
Computes a division with remainder.
Returns a tuple `T,Rem,U` where `matrix(I)*matrix(U)=matrix(G)*matrix(T)+matrix(Rem)`.
```

File: src/ideal/ideal.jl (hunk 1 above)
Comment (fingolfin) [reply src/ideal/ideal.jl:653]: This seems underspecified -- returning three zero matrices would satisfy this definition, wouldn't it?

Comment (tthsqe12) [reply src/ideal/ideal.jl]: I used the documentation for `lift` to hopefully provide a better doc string now.

**Pull request merged**